File: 4.sort/2545.SortTheStudentsByTheirKthScore/main.py
from typing import List

# https://leetcode.com/problems/sort-the-students-by-their-kth-score/

# TC: O(mlogm), SC: O(1)
class Solution:
    def sortTheStudents(self, score: List[List[int]], k: int) -> List[List[int]]:
        score.sort(key=lambda x: x[k], reverse=True)
        return score

File: 4.sort/2545.SortTheStudentsByTheirKthScore/main.go
package main

import "sort"

// https://leetcode.com/problems/sort-the-students-by-their-kth-score/

func sortTheStudents(score [][]int, k int) [][]int {
	sort.SliceStable(score, func(i, j int) bool {
		return score[i][k] > score[j][k]
	})

	return score
}
